system/Publisher/ContentReplacer.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Publisher;
+
+class ContentReplacer
+{
+    /**
+     * Replace content
+     *
+     * @param array $replaces [search => replace]
+     */
+    public function replace(string $content, array $replaces): string
+    {
+        return strtr($content, $replaces);
+    }
+
+    /**
+     * Add text
+     *
+     * @param string $text    Text to add.
+     * @param string $pattern Regexp search pattern.
+     * @param string $replace Regexp replacement including text to add.
+     *
+     * @return bool|string true: already updated, false: regexp error.
+     */
+    public function add(string $content, string $text, string $pattern, string $replace)
```

Comment:
Any reason this needs to be public?
```suggestion
    private function add(string $content, string $text, string $pattern, string $replace)
```